vassal-app/src/main/java/VASSAL/build/module/map/KeyBufferer.java

```diff
@@ -172,7 +169,7 @@ else if (!s.isExpanded()) {
         kbuf.clear();
         
         //BR// This section allows band-select to be attempted from non-moving pieces w/o preventing click-to-select from working 
-        if (bandSelect == BandSelectType.SPECIAL && p != null && !ignoreEvent) {
```

Comment:
How do you know you don't need to check these here?

Reply:
`bandSelect` can only have the value `SPECIAL` if:

1. the `p != null` check in line 112 resolved to `true` and `bandSelect` was set to `SPECIAL` in line 116
2. the `if (p != null && !ignoreEvent)` in line 121 resolved to `true` otherwise `bandSelect` would have been set to `NORMAL` in the else-block in line 163

Also, this whole method is way too long at 85 lines.

Reply:
I would like to leave this alone for context until the whole thing can be refactored.